### PR TITLE
Add contributing.md

### PR DESCRIPTION
**Pull request by weshayutin**
## Why the changes were made

OADP did not yet have a contributing guide.

## How to test the changes made

READ :)

### CODE REVIEW

**Comment by weshayutin** [comment]
contributing - https://github.com/openshift/oadp-operator/pull/1586
code of conduct - https://github.com/openshift/oadp-operator/pull/1576

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Contribution Guidelines
+
+### Pull Request guidelines
+1. Follow the PR Templates
+   - Use PR Template instructions, they allow to ensure consistency and clarity.
+1. Avoid Self-Overrides
+   - If automated tests fail (e.g., Prow tests), ask reviewers to override tests if necessary.
```
Comment (kaovilai) [inline docs/CONTRIBUTING.md:7]: Can reviewers override tests without author asking?

Comment (weshayutin) [reply docs/CONTRIBUTING.md:7]: updated :) thanks

**Comment by kaovilai** [comment]
/override ci/prow/4.17-ci-index

This is not relevant to .md

**Review by shubham-pampattiwar — APPROVED** [review]

**Review by kaovilai — APPROVED** [review]
/lgtm

**Pull request merged**